pkg/provider/vault/symlink.go

```diff
@@ -0,0 +1,43 @@
+package vault
+
+import (
+	"context"
+	"regexp"
+	"strings"
+)
+
+func isSymlink(secret any) bool {
+	return strings.HasPrefix(secret.(string), "vault://")
+}
+
+func extractSymlinkParts(secret any) (paramsMap map[string]string) {
+	r := regexp.MustCompile(`vault://(?P<Path>.*)#(?P<Secret>\w+)(@(?P<Version>.*)?)?`)
```

Comment:
are there any rules for what can be a path, a secret and version? Should the regex be more specific to only valid vault objects?